sklearn/metrics/pairwise.py

```diff
@@ -1421,6 +1421,12 @@ def pairwise_distances(X, Y=None, metric="euclidean", n_jobs=None, **kwds):
                             " support sparse matrices.")
 
         dtype = bool if metric in PAIRWISE_BOOLEAN_FUNCTIONS else None
+
+        if dtype == bool and (X.dtype != bool or Y.dtype != bool):
+            msg = ("Data was converted to boolean "
+                   "for metric %s" % (metric))
```

Comment:
Cosmetics: this fits on one line and there is no need to wrap the metric variable with parenthesis:

```python
            msg = "Data was converted to boolean for metric %s" % metric
```

Reply:
done